ethereum/core/src/main/java/tech/pegasys/pantheon/ethereum/mainnet/MainnetProtocolSpecs.java

```diff
@@ -285,12 +286,13 @@ private MainnetProtocolSpecs() {}
       final OptionalInt configContractSizeLimit,
       final OptionalInt configStackSizeLimit,
       final boolean enableRevertReason) {
+    checkArgument(chainId.isPresent(), "Istanbul requires the use of chainId");
```

Comment:
If you trace back to where the `chainId` is set, it looks like it's only ever missing in test cases.  Might be worth going through and making the `chainId` argument a plain unwrapped `BigInteger`, in which case we wouldn't need this check.

Reply:
FixedDifficultyProtocolSchedule uses it too.  It's at least a 4 file change, larger than this PR, so that level of change would deserve its own PR.

Reply:
There's no requirement that a chain has a `chainId` - it may be omitted from the genesis config.  This is a slightly interesting hole in the ChainID opcode spec. Doesn't apply to any public chain but there may be chains out there it does affect.  So maybe ChainID opcode should return 0 when there is no chain ID?